packages/shared-base/src/utils/subscription.ts

```diff
@@ -1,8 +1,21 @@
 import { noop } from 'lodash-unified'
 import type { ValueRef } from '@dimensiondev/holoflows-kit'
-import type { Subscription } from 'use-subscription'
 import { None, Option, Some } from 'ts-results'
 
+type Unsubscribe = () => void
+
+export interface Subscription<T> {
+    /**
+     * (Synchronously) returns the current value of our subscription.
+     */
+    getCurrentValue: () => T
```

Comment:
I think you should add a `getServerValue?(): T` to `Subscription` type, and export the following hooks from the `shared` (React hooks should not inside the `shared-base`) package:

```js
function useSubscription<T>(sub: Subscription<T>): T {
    return useSyncExternalStoreWithSelector(
        sub.subscribe,
        sub.getCurrentValue,
        sub.getServerValue|| sub.getCurrentValue,
        (s) => s,
    )
}
```

It will make the migration easier.

Reply:
We haven't made a plan to use a full-featured store like zustand or redux, thus I think easy migration is more important.

> What happened if different logic for getServerValue would be applied across different useSyncExternalStoreWithSelector usage in the future?

Can you explain this part?